packages/editor/src/lib/editor/Editor.ts

```diff
@@ -8552,325 +8569,357 @@ export class Editor extends EventEmitter<TLEventMap> {
 	 * ```
 	 *
 	 * @param info - The event info.
-	 *
+	 h*
 	 * @public
 	 */
 	dispatch = (info: TLEventInfo): this => {
+		if (this._pendingEventsForNextTick.length === 0) {
+			this._pendingEventsForNextTick.push(info)
+			if (this._isCoalesableEvent(info)) {
+				this._allEventsSinceLastTick.push(info)
+			}
+		} else {
+			if (this._shouldCoallesce(info)) {
+				this._pendingEventsForNextTick[0] = info
+				if (this._isCoalesableEvent(info)) {
+					this._allEventsSinceLastTick.push(info)
+				}
+			} else {
+				// Event has changed. We flush the currently pending events
+				this._flushEventsForTick(0)
+				// Then we add the new event to the queue (flushing clears the queue)
+				this._pendingEventsForNextTick = [info]
+			}
+		}
+		return this
+	}
+
+	private _pendingEventsForNextTick: TLEventInfo[] = []
+	private _allEventsSinceLastTick: TLPointerMoveEventInfo[] = []
+
+	private _flushEventsForTick = (elapsed: number) => {
+		this.batch(() => {
+			if (this._pendingEventsForNextTick.length > 0) {
+				const events = [...this._pendingEventsForNextTick]
+				this._pendingEventsForNextTick.length = 0
+				for (const info of events) {
+					this._flushEventForTick(info)
+				}
+			}
+			this.root.handleEvent({ type: 'misc', name: 'tick', elapsed })
+			this.scribbles.tick(elapsed)
+		})
+		this._allEventsSinceLastTick.length = 0
+	}
+
+	private _flushEventForTick = (info: TLEventInfo) => {
 		// prevent us from spamming similar event errors if we're crashed.
 		// todo: replace with new readonly mode?
 		if (this.getCrashingError()) return this
 
 		const { inputs } = this
 		const { type } = info
 
-		this.batch(() => {
```

Comment:
Most of the changes below are due to the indentation change. This batch was moved up to ` _flushEventsForTick`